src/coreclr/src/jit/emitarm64.cpp

```diff
@@ -10790,54 +11042,88 @@ void emitter::emitDispVectorReg(regNumber reg, insOpts opt, bool addComma)
         printf(", ");
 }
 
-/*****************************************************************************
- *
- *  Display an vector register index suffix
- */
+//------------------------------------------------------------------------
+// emitDispVectorRegIndex: Display a SIMD vector register name with element index
+//
 void emitter::emitDispVectorRegIndex(regNumber reg, emitAttr elemsize, ssize_t index, bool addComma)
 {
     assert(isVectorRegister(reg));
     printf(emitVectorRegName(reg));
+    emitDispElemsize(elemsize);
+    printf("[%d]", index);
 
-    switch (elemsize)
+    if (addComma)
+        printf(", ");
+}
+
+//------------------------------------------------------------------------
+// emitDispVectorRegList: Display a SIMD vector register list
+//
+void emitter::emitDispVectorRegList(regNumber firstReg, unsigned listSize, insOpts opt, bool addComma)
+{
+    assert(isVectorRegister(firstReg));
+
+    regNumber currReg = firstReg;
+
+    printf("{");
+    for (unsigned i = 0; i < listSize; i++)
     {
-        case EA_1BYTE:
-            printf(".b");
-            break;
-        case EA_2BYTE:
-            printf(".h");
-            break;
-        case EA_4BYTE:
-            printf(".s");
-            break;
-        case EA_8BYTE:
-            printf(".d");
-            break;
-        default:
-            assert(!"invalid elemsize");
-            break;
+        const bool notLastRegister = (i != listSize - 1);
+        emitDispVectorReg(currReg, opt, notLastRegister);
+        currReg = (currReg == REG_V31) ? REG_V0 : REG_NEXT(currReg);
     }
-
-    printf("[%d]", index);
+    printf("}");
 
     if (addComma)
+    {
         printf(", ");
+    }
 }
 
-/*****************************************************************************
- *
- *  Display an arrangement suffix
- */
-void emitter::emitDispArrangement(insOpts opt)
+//------------------------------------------------------------------------
+// emitDispVectorElemList: Display a SIMD vector element list
+//
+void emitter::emitDispVectorElemList(
```

Comment:
Would this be clearer as `emitDispVectorRegListWithSize`? As named, I wasn't sure the difference between `emitDispVectorRegList` and `emitDispVectorElemList` -- namely, what is a "VectorElem"?

Reply:
These terminology from *C1.2.5 Register names* - I don't have a strong preference how to name the function - I tried to follow what the Arm docs said.